ksql-parser/src/main/java/io/confluent/ksql/parser/tree/Sink.java

```diff
@@ -51,19 +53,29 @@ public static Sink of(
       final CreateSourceAsProperties properties,
       final Optional<Expression> partitionBy
   ) {
-    final Optional<ColumnName> partitionByExp = partitionBy
-        .map(Object::toString)
-        .map(String::toUpperCase)
-        .map(ColumnName::of);
+    if (partitionBy.isPresent()) {
+      final Expression partitionByExp = partitionBy.get();
+      if (partitionByExp instanceof ColumnReferenceExp) {
+        return new Sink(
+            name,
+            createSink,
+            properties,
+            Optional.of(((ColumnReferenceExp) partitionByExp).getReference())
+        );
+      } else {
+        throw new KsqlException(
+            "Expected partition by to be a valid column but got " + partitionByExp);
+      }
```

Comment:
nit: `if return else throw` is a bit of a anti-pattern.  How about just `if (not thing) {throw}`? Or at the very least get rid of the unnecessary `else`.